src/test/java/org/folio/rest/api/ItemEffectiveLocationTest.java

```diff
@@ -220,72 +182,135 @@ public void canCalculateEffectiveLocationOnHoldingUpdateWhenSomeItemsHasLocation
-        .put("temporaryLocationId", onlineLocationId.toString())
-    );
-
-    assertEquals(getItem(item.getId()).getEffectiveLocationId(), onlineLocationId.toString());
+  @Parameters(method = "parameters")
+  public void canCalculateEffecticeLocationOnItemUpdate(
+      int holdingPerm, int holdingTemp, int itemPerm, int itemTemp, int newItemPerm, int newItemTemp) throws Exception {
+
+    UUID holdingsRecordId = createHolding(instanceId, location[holdingPerm], location[holdingTemp]);
+
+    Item item = buildItem(holdingsRecordId, location[itemPerm], location[itemTemp]);
+    UUID itemId = UUID.fromString(item.getId());
+    IndividualResource itemResource = createItem(item);
+    JsonObject item2 = itemResource.getJson();
+    assertEffectiveLocation(item2, holdingPerm, holdingTemp, itemPerm, itemTemp);
+
+    setPermTempLocation(item2, newItemPerm, newItemTemp);
+    itemsClient.replace(itemId, item2);
+    JsonObject item3 = itemsClient.getById(itemId).getJson();
+    assertEffectiveLocation(item3, holdingPerm, holdingTemp, newItemPerm, newItemTemp);
   }
 
   @Test
-  public void canCalculateEffectiveLocationOnItemAllLocationUpdate() throws Exception {
-    UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId, annexLibraryLocationId);
-
-    Item item = buildItem(holdingsRecordId, null, null);
-    createItem(item);
-
-    Item itemFetched = getItem(item.getId());
-    assertEquals(itemFetched.getEffectiveLocationId(), annexLibraryLocationId.toString());
-
-    itemsClient.replace(UUID.fromString(itemFetched.getId()),
-      JsonObject.mapFrom(itemFetched).copy()
-        .put("temporaryLocationId", onlineLocationId.toString())
-        .put("permanentLocationId", secondFloorLocationId.toString())
-    );
-
-    assertEquals(getItem(item.getId()).getEffectiveLocationId(), onlineLocationId.toString());
+  @Parameters(method = "parameters")   // res-use swapping item and holding
+  public void canCalculateEffecticeLocationOnHoldingsUpdate(
+      int itemPerm, int itemTemp, int holdingPerm, int holdingTemp, int newHoldingPerm, int newHoldingTemp) throws Exception {
+
+    UUID holdingsRecordId = createHolding(instanceId, location[holdingPerm], location[holdingTemp]);
+
+    Item item = buildItem(holdingsRecordId, location[itemPerm], location[itemTemp]);
+    UUID itemId = UUID.fromString(item.getId());
+    JsonObject item2 = createItem(item).getJson();
+    assertEffectiveLocation(item2, holdingPerm, holdingTemp, itemPerm, itemTemp);
+
+    JsonObject holding2 = holdingsClient.getById(holdingsRecordId).getJson();
+    setPermTempLocation(holding2, newHoldingPerm, newHoldingTemp);
+    holdingsClient.replace(holdingsRecordId, holding2);
+    JsonObject item3 = itemsClient.getById(itemId).getJson();
+    assertEffectiveLocation(item3, newHoldingPerm, newHoldingTemp, itemPerm, itemTemp);
   }
 
   @Test
   public void canSearchItemByEffectiveLocation() throws Exception {
+    StorageTestSuite.deleteAll(itemsStorageUrl(""));
```

Comment:
This is the only test that requires all holdings and items been deleted.
All other tests don't.
There are two methods that run 112 times because they take the 112 data sets from the parameters method. In some former version there were many more data sets and they took 30 seconds.
Having deleteAll only in this method saves 2*112=224 delete requests per table.

Reply:
I can understand wanting to reduce the amount of record clean up needed. Maybe this test could be moved to a test class better suited to that, away from the tests that need considerable record set up?

Reply:
Out of interest, why do we need 224 tests to check this behaviour?

Reply:
@julianladisch are you going to move it to a separate test class?